pkg/controller/hawtio/hawtio_controller.go

```diff
@@ -297,13 +305,44 @@ func (r *ReconcileHawtio) Reconcile(request reconcile.Request) (reconcile.Result
 	}
 
 	if isOpenShift4 {
-		// Check whether client certificate secret exists
+		//Check whether client certificate secret exists
 		_, err := r.coreClient.Secrets(request.Namespace).Get(ctx, request.Name+"-tls-proxying", metav1.GetOptions{})
+		//TODO add automated regeneration of certificate after the expiration
 		if err != nil {
 			if errors.IsNotFound(err) {
-				reqLogger.Info("Client certificate secret must be created", "secret", request.Name+"-tls-proxying")
-				// Let's poll for the client certificate secret to be created
-				return reconcile.Result{Requeue: true, RequeueAfter: 5 * time.Second}, nil
+				secretName := request.Name + "-tls-proxying"
+				reqLogger.Info("Client certificate not found, certificate will be created now", "secret", secretName)
+
+				caSecret, err := r.coreClient.Secrets("openshift-service-ca").Get(ctx, "signing-key", metav1.GetOptions{})
+				if err != nil {
+					reqLogger.Error(err, "Reading certificate authority signing key failed", err)
+					return reconcile.Result{}, err
+				}
+
+				if hawtio.Spec.CertificateConfig.CommonName == "" {
+					if r.CertificateCommonName == "" {
+						hawtio.Spec.CertificateConfig.CommonName = "hawtio-online.hawtio.svc"
+					} else {
+						hawtio.Spec.CertificateConfig.CommonName = r.CertificateCommonName
+					}
+				}
```

Comment:
Let's not update the Hawtio spec, but rather apply the defaults in the `generateCertificateSecret` method.

Reply:
will do that